enterprise/frontend/src/metabase-enterprise/metabot/components/MetabotAdmin/MetabotAdminPurchase.tsx

```diff
@@ -0,0 +1,234 @@
+import type { FetchBaseQueryError } from "@reduxjs/toolkit/query";
+import { useCallback, useState } from "react";
+import { t } from "ttag";
+import * as Yup from "yup";
+
+import {
+  SettingsPageWrapper,
+  SettingsSection,
+} from "metabase/admin/components/SettingsSection";
+import { usePurchaseMetabotCloudAddOnMutation } from "metabase/api/metabot";
+import ExternalLink from "metabase/common/components/ExternalLink";
+import {
+  Form,
+  FormCheckbox,
+  FormProvider,
+  FormSubmitButton,
+} from "metabase/forms";
+import {
+  Box,
+  Button,
+  Divider,
+  Flex,
+  Group,
+  List,
+  Loader,
+  Modal,
+  type ModalProps,
+  Stack,
+  Text,
+  Title,
+} from "metabase/ui";
+
+// https://redux-toolkit.js.org/rtk-query/usage/error-handling
+// https://redux-toolkit.js.org/rtk-query/usage-with-typescript#type-safe-error-handling
+const isFetchBaseQueryError = (error: unknown): error is FetchBaseQueryError =>
+  error instanceof Object && "status" in error && "data" in error;
+
+type IFieldError =
+  | string
+  | {
+      message: string;
+    }
+  | {
+      errors: { [key: string]: any };
+    };
+
+const isFieldError = (error: unknown): error is IFieldError =>
+  typeof error === "string" ||
+  (error instanceof Object &&
+    (("message" in error && typeof error.message === "string") ||
+      ("errors" in error &&
+        error.errors instanceof Object &&
+        "terms_of_service" in error.errors &&
+        typeof error.errors.terms_of_service === "string")));
+
+export const handleFieldError = (error: unknown) => {
+  if (isFieldError(error)) {
+    if (typeof error === "string") {
+      throw { data: { errors: { terms_of_service: error } } };
+    } else if ("message" in error) {
+      throw { data: { errors: { terms_of_service: error.message } } };
+    } else if ("errors" in error) {
+      throw { data: error };
+    }
+  }
```

Comment:
Nitpick:
We could avoid nested if`s
```suggestion
  if (!isFieldError(error)) {
    return;
  }

  if (typeof error === "string") {
    throw { data: { errors: { terms_of_service: error } } };
  }
  
  if ("message" in error) {
    throw { data: { errors: { terms_of_service: error.message } } };
  }
  
  if ("errors" in error) {
    throw { data: error };
  }
```